src/CommitmentsV2/SFA.DAS.CommitmentsV2/Models/Commitment.cs

```diff
@@ -118,24 +118,28 @@ private IEnumerable<DomainError> BuildCostValidationFailures(DraftApprenticeship
 
         private IEnumerable<DomainError> BuildReferenceValidationFailures(DraftApprenticeshipDetails draftApprenticeshipDetails)
         {
-            if (EditStatus == EditStatus.ProviderOnly)
-            {
-                if (draftApprenticeshipDetails.Reference != null && draftApprenticeshipDetails.Reference.Length > 20)
-                {
-                    yield return new DomainError(nameof(draftApprenticeshipDetails.Reference), "The Reference must be 20 characters or fewer");
-                }
-            }
-            else
-            {
             if (draftApprenticeshipDetails.Reference != null && draftApprenticeshipDetails.Reference.Length > 20)
             {
                 yield return new DomainError(nameof(draftApprenticeshipDetails.Reference), "The Reference must be 20 characters or fewer");
             }
+        }
+        private IEnumerable<DomainError> BuildUlnValidationFailures(DraftApprenticeshipDetails draftApprenticeshipDetails, IUlnValidator ulnValidator)
+        {
+            if (!string.IsNullOrWhiteSpace(draftApprenticeshipDetails.Uln))
+            {
+                var validationResult = ulnValidator.Validate(draftApprenticeshipDetails.Uln);
+                switch(validationResult)
+                {
+                    case UlnValidationResult.IsInValidTenDigitUlnNumber:
+                        yield return new DomainError(nameof(draftApprenticeshipDetails.Uln), "You must enter a 10-digit unique learner number");
+                        yield break;
```

Comment:
is yield break needed here. Surely the "yield return " would just be needed

Reply:
you have to break or yield break out of the switch